packages/salesforcedx-vscode-core/src/orgBrowser/metadataCmp.ts

```diff
@@ -136,52 +137,45 @@ export class ComponentUtils {
     const describeSObjectFields = await connection.describe(sObjectName);
     const describeSObjectFieldsList = describeSObjectFields.fields;
     const result = {status: 0, result: describeSObjectFieldsList};
-    const jsonResult = JSON.stringify(result);
+    const jsonResult = JSON.stringify(result, null, 2);
     fs.writeFileSync(componentsPath, jsonResult);
     return jsonResult;
   }
 
   public async loadComponents(
     defaultOrg: string,
     metadataType: string,
-    sObject?: string,
+    folderName?: string,
     forceRefresh?: boolean
   ): Promise<string[]> {
     const componentsPath = await this.getComponentsPath(
       metadataType,
       defaultOrg,
-      sObject
+      folderName
     );
 
     let componentsList: string[];
     const connection = await workspaceContext.getConnection();
 
-    if (metadataType === 'CustomObject' && sObject) {
+    if (metadataType === 'CustomObject' && folderName) {
       if (forceRefresh || !fs.existsSync(componentsPath)) {
-        componentsList = await this.fetchCustomObjectsFields(sObject, connection, componentsPath);
+        componentsList = await this.fetchCustomObjectsFields(folderName, connection, componentsPath);
       } else {
-        componentsList = this.fetchExistingCustomObjectsFields(sObject, componentsPath);
+        componentsList = this.fetchExistingCustomObjectsFields(folderName, componentsPath);
       }
     } else {
       if (forceRefresh || !fs.existsSync(componentsPath)) {
-        componentsList = await this.fetchMetadataComponents(metadataType, connection, componentsPath);
+        componentsList = await this.fetchMetadataComponents(metadataType, connection, componentsPath, folderName);
```

Comment:
The PR looks great. Just one suggestion - Can you change metadataCmp.test.ts to test the arguments passed to fetchMetadataComponentsStub? Right now we are just testing fetchMetadataComponentsStub and other stubs are being called and not the arguments. Everything else looks great 👍

Reply:
@jag-sfdc I have added tests to check for the arguments with which these functions are getting called. Also added additional tests to test the code-flow when the folders within `Dashboards` metadata component are clicked.